nnvm/python/nnvm/frontend/onnx.py

```diff
@@ -824,6 +824,22 @@ def from_onnx(self, graph, opset):
             else:
                 self._num_input += 1
                 self._nodes[i_name] = _sym.Variable(name=i_name)
+        # get list of unsupported ops
+        convert_map = _get_convert_map(opset)
+        unsupported_ops = set()
+        for node in graph.node:
+            op_name = node.op_type
+            if op_name not in convert_map and \
+               op_name != 'Constant' and \
+               op_name not in _identity_list:
+                unsupported_ops.add(op_name)
+        if unsupported_ops:
+            msg = ['The following operators are not supported for frontend ONNX: ']
+            for i, op_name in enumerate(unsupported_ops):
```

Comment:
how about` ops = ', '.join(str(op) for op in unsupported_ops)`?

Reply:
We’d still have to concatenate the ‘The following ops...’ string and I generally avoid concatenating strings with +. Your way is neater, though, so I’ll change it.